### ISSUE TITLE
Exception setting up ContentType

### ISSUE THREAD

**Issue by eduardomart** [bolt/bolt#7040]
I tried using the example in the [3.4 docs](url=https://docs.bolt.cm/3.4/howto/singleton-contenttype#):
```
settings:
    name: Site Settings
    singular_name: Site Settings
    fields:
        headers:
            type: repeater
            group: headers
            fields:
                header_title:
                    type: text
                    label: Title text
        footer_text:
            type: html
            group: footer
        footer_copyright:
            type: text
        company_address:
            type: geolocation
            group: common
    viewless: true
    singleton: true
    default_status: published
    sort: -id
    icon_many: "fa:cogs"
    icon_one: "fa:cog"
```

Saved and right after, anything in the backend will display this:

![2017-09-30 14-18-56](https://user-images.githubusercontent.com/1833020/31049571-a19652d8-a5ea-11e7-9ce0-c6c6f82f349c.png)

Details
-------

 - Relevant Bolt Version: [ 3.4 beta 1 ]
 - PHP version: [ 7.1 ]
 - Used web server: [ Apache ]

**Comment by GawainLynch** [comment]
Are you sure they are the settings you're using? The exception is coming from a ContentType called `mysettings`

**Comment by eduardomart** [comment]
Yeah I tried the first one `mysettings` using `singleton: true` and then `viewless: true` thinking I was missing something I copied and pasted the example in the 3.4 docs but I am still getting the error with the same `mysettings` ct name

**Comment by GawainLynch** [comment]
… and you've updated the database? (which is what the exception is, not that it is a singleton)

**Comment by GawainLynch** [comment]
Fixed in #7041
